Vector/Views/EncryptionInfoView/EncryptionInfoView.m

```diff
@@ -0,0 +1,359 @@
+                
+                [deviceInformationString appendAttributedString:[[NSMutableAttributedString alloc]
+                                                                 initWithString:NSLocalizedStringFromTable(@"room_event_encryption_info_device_verification", @"Vector", nil)
+                                                                 attributes:@{NSForegroundColorAttributeName : kVectorTextColorBlack,
+                                                                              NSFontAttributeName: [UIFont boldSystemFontOfSize:14]}]];
+                [deviceInformationString appendAttributedString:verification];
+                [deviceInformationString appendAttributedString:[EncryptionInfoView verticalWhitespace]];
+                
+                [deviceInformationString appendAttributedString:[[NSMutableAttributedString alloc]
+                                                                 initWithString:NSLocalizedStringFromTable(@"room_event_encryption_info_device_fingerprint", @"Vector", nil)
+                                                                 attributes:@{NSForegroundColorAttributeName : kVectorTextColorBlack,
+                                                                              NSFontAttributeName: [UIFont boldSystemFontOfSize:14]}]];
+                [deviceInformationString appendAttributedString:[[NSMutableAttributedString alloc]
+                                                                 initWithString:fingerprint
+                                                                 attributes:@{NSForegroundColorAttributeName : kVectorTextColorBlack,
+                                                                              NSFontAttributeName: [UIFont systemFontOfSize:14]}]];
+                [deviceInformationString appendAttributedString:[EncryptionInfoView verticalWhitespace]];
+            }
+            else
+            {
+                // Unknown device
+                [deviceInformationString appendAttributedString:[[NSMutableAttributedString alloc]
+                                                                 initWithString:NSLocalizedStringFromTable(@"room_event_encryption_info_device_unknown", @"Vector", nil)
+                                                                 attributes:@{NSForegroundColorAttributeName : kVectorTextColorBlack,
+                                                                              NSFontAttributeName: [UIFont italicSystemFontOfSize:14]}]];
+            }
+            
+            
+            [textViewAttributedString appendAttributedString:deviceInformationString];
+            
+            self.textView.attributedText = textViewAttributedString;
+        }
+        else
+        {
+            _textView.text = nil;
+        }
+    }
+    
+    return self;
+}
+
+- (void)dealloc
+{
+    mxEvent = nil;
+    mxSession = nil;
+}
+
++ (NSAttributedString *)verticalWhitespace
+{
+    @synchronized(self)
```

Comment:
Why is there a @synchronized whereas everything is on the UI thread?
Did you mean dispatch_once?